packages/astro/src/core/cookies/cookies.ts

```diff
@@ -2,17 +2,156 @@ import type { CookieSerializeOptions } from 'cookie';
 import { parse, serialize } from 'cookie';
 import { AstroError, AstroErrorData } from '../errors/index.js';
 
-export type AstroCookieSetOptions = Pick<
-	CookieSerializeOptions,
-	'domain' | 'path' | 'expires' | 'maxAge' | 'httpOnly' | 'sameSite' | 'secure' | 'encode'
->;
+// `AstroCookieSetOptions` and `AstroCookieDeleteOptions` are manually copied subsections of `CookieSerializeOptions` because `Omit` and `Pick` result in all fields being required.
```

Comment:
Any reason it cannot be `Partial<Pick...`?

Reply:
I think it's because jsdoc annotations are lost anyway